vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -447,8 +407,14 @@ def acquire_read(self,
                     # if this block is not ready,
                     # we need to wait until it is written
 
-                    # Release the processor to other threads
-                    self._read_spin_timer.spin()
+                    remaining = None if timeout is None else max(
+                        0.0, timeout - (time.monotonic() - start_time))
+                    if remaining is None or remaining > 0.0:
+                        poll_timeout_ms = 50
+                        if remaining is not None:
+                            poll_timeout_ms = min(poll_timeout_ms,
+                                                  int(remaining * 1000))
+                        self.local_socket.poll(timeout=poll_timeout_ms)
```

Comment:
Simplify the logic a bit?
```suggestion
                    poll_timeout_ms = 50
                    if timeout is not None:
                        remaining = start_time + timeout - time.monotonic()
                        poll_timeout_ms = min(int(remaining * 1000),
                                              poll_timeout_ms)
                    if poll_timeout_ms > 0:
                        self.local_socket.poll(timeout=poll_timeout_ms)
```